core/rpc_client.py

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env python
+#  -*- coding: utf-8 -*-
+# *********************************************************************
+# plankton - a library for creating hardware device simulators
+# Copyright (C) 2016 European Spallation Source ERIC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# *********************************************************************
+
+import zmq
+import uuid
+import types
+
+try:
+    import exceptions
+except ImportError:
+    import builtins as exceptions
+
+
+class JSONRPCServerSideException(Exception):
+    def __init__(self, type, message):
+        """
+        This exception type replaces exceptions that are raised on the server,
+        but unknown (i.e. not in the exceptions-module) on the client side.
+        To retain as much information as possible, the exception type on the server and
+        the message are stored.
+
+        :param type: Type of the exception on the server side.
+        :param message: Exception message.
+        """
+        self.type = type
+        self.message = message
+
+    def __str__(self):
+        return 'Exception on server side of type \'{}\': \'{}\''.format(self.type, self.message)
```

Comment:
Similar to the comment on the other exception, you could do just a `pass` (and expect the user to format their own message). Alternatively, if you want to keep the usage as-is:

```python
class JSONRPCServerSideException(Exception):
    def __init__(self, type, message):
        """docstring"""
        super(JSONRPCServerSideException, self).__init__('Exception on server side of type \'{}\': \'{}\''.format(self.type, self.message))
```

Optionally, could also keep the `self.type` and `self.message` assignment after the `super` if you want them accessible.